File: src/main/java/com/arnas/klatrebackend/interfaces/repositories/PlaceRepository.java
package com.arnas.klatrebackend.interfaces.repositories;

import com.arnas.klatrebackend.dataclasses.Place;
import com.arnas.klatrebackend.dataclasses.PlaceUpdateDTO;
import org.springframework.lang.NonNull;

import java.util.Optional;

public interface PlaceRepository {
    Place[] getPlacesByGroupId(long groupId);
    long addPlaceToGroup(long groupId, String name, @NonNull String description);
    Optional<Place> getPlaceById(long placeId);
    int updatePlace(long placeId, String name, String description, Long groupId, Long gradingSystem);
    int updatePlace(PlaceUpdateDTO placeUpdateDTO);
    int deletePlace(Long placeId);
}

File: src/main/kotlin/com/arnas/klatrebackend/repository/PlaceRepository.kt
package com.arnas.klatrebackend.repository

import com.arnas.klatrebackend.dataclass.Place
import com.arnas.klatrebackend.interfaces.repositories.PlaceRepositoryInterface
import org.springframework.jdbc.core.RowMapper
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
import org.springframework.stereotype.Repository
import java.sql.ResultSet

@Repository
class PlaceRepository(
    private var jdbcTemplate: NamedParameterJdbcTemplate
): PlaceRepositoryInterface {

    override fun getPlacesByGroupId(groupId: Long): List<Place> {

        val rowMapper = RowMapper<Place> { rs: ResultSet, _: Int ->
            Place(
                id = rs.getLong("id"),
                name = rs.getString("name"),
                description = rs.getString("description"),
                groupID = rs.getLong("group_id"),
                gradingSystem = rs.getLong("grading_system_id")
            )
        }
        val places =  jdbcTemplate.query("SELECT * FROM places WHERE group_id = :groupId",
            mapOf("groupId" to groupId),
            rowMapper
        )
        return places.toList()
    }
}